src/components/selectYear.js

```diff
@@ -0,0 +1,112 @@
+import React, { PureComponent, PropTypes } from 'react'
+import { connect } from 'react-redux'
+import * as actions from '../actions/actionIndex'
+import reactCSS from 'reactcss'
+
+const styles = reactCSS({
+  'default': {
+    button: {
+      color: 'orange',
+      fontSize: 23,
+      padding: 15,
+      backgroundColor: 'rgba(0, 0, 0, 0)',
+      border: 0,
+      paddingTop: 5,
+    }
+  }
+})
+
+const { object, func} = PropTypes
+
+class SelectYear extends PureComponent {
+
+  state = {
+    historyYear: 1977
+  }
+
+  static PropTypes = {
+    historyYear: object,
+    goToFollowingYear: func,
+    loadHistoryData: func,
+  }
+
+  static defaultProps = {
+    historyYear: 1977,
+  }
+
+  validateYear = (year) => {
+    if (year > 2016) {
+      alert('Please select a year prior to 2017!')
+      return false
+    }
+    else if (year < 1949) {
+      alert('Please select a year later than 1949!')
+      return false
+    }
+    else return true
+  }
+
+  gotoNextYear = () => {    
+    const nextYear = this.state.historyYear + 1
+    const valid = this.validateYear(nextYear)
+    if (valid === true) {
```

Comment:
I refactored it this way:
`gotoYear = (differenceValue) => {
    const year = this.props.historyYear + differenceValue
    const valid = this.validateYear(year)
    if (valid) {
      this.setState({historyYear: year})
      this.props.updateHistoryYear(year)
      this.props.loadHistoryData(year)
    }
  }

  gotoNextYear = () => {    
    this.gotoYear(1)
  }
  gotoTenYearsLater = () => {    
    this.gotoYear(10)
  }
  gotoLastYear = () => {
    this.gotoYear(-1)
  }
  gotoTenYearsAgo = () => {
    this.gotoYear(-10)
  }

  render() {
    return (    
       <div>
        <span><input style={ styles.button } type='submit' value='<<<' onClick={this.gotoTenYearsAgo}/></span>
        <span><input style={ styles.button } type='submit' value='<' onClick={this.gotoLastYear}/></span>
        <span style={ styles.button }> {this.props.historyYear} </span>
        <span><input style={ styles.button } type='submit' value='>' onClick={this.gotoNextYear}/></span>
        <span><input style={ styles.button } type='submit' value='>>>' onClick={this.gotoTenYearsLater}/></span>
      </div>
    )
  }`

But when I tried `gotoYear = (differenceValue) => {
    const year = this.props.historyYear + differenceValue
    const valid = this.validateYear(year)
    if (valid) {
      this.setState({historyYear: year})
      this.props.updateHistoryYear(year)
      this.props.loadHistoryData(year)
    }
  }

  render() {
    return (    
       <div>
        <span><input style={ styles.button } type='submit' value='<<<' onClick={this.gotoYear(-10)}/></span>
        ......
      </div>
    )
  }`
It didn't work. Maybe we can talk about it in our next session.